test/test_per_overload_api.py

```diff
@@ -0,0 +1,64 @@
+import torch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch.testing._internal.logging_tensor import LoggingTensor, log_input, capture_logs, no_dispatch
+from torch.utils._pytree import tree_map
+from torch.utils._python_dispatch import enable_python_mode
+import copy
+
+class TestPerOverloadAPI(TestCase):
```

Comment:
Could we test that per-overload op is usable in a TorchScripted function?

Reply:
No fails with:
`RuntimeError: attribute lookup is not defined on builtin`
probably because TorchScript allows attribute lookups only on modules and OpOverloadPacket is not a module.

Reply:
full error log:

```
  File "/fsx/users/chourdiaanjali/tree1/test/test_per_overload.py", line 22, in test_torch_aten_add
    self.checkScript(fn, (torch.randn(2), torch.randn(2)))
  File "/fsx/users/chourdiaanjali/tree1/torch/testing/_internal/jit_utils.py", line 482, in checkScript
    self.checkScript(
  File "/fsx/users/chourdiaanjali/tree1/torch/testing/_internal/jit_utils.py", line 466, in checkScript
    cu = torch.jit.CompilationUnit(script, _frames_up=frames_up)
RuntimeError:
attribute lookup is not defined on builtin:
  File "<string>", line 2
def fn(a, b):
    return torch.ops.aten.add.Tensor(a, b)
```

Reply:
Is per-overload op intended an eager-only feature?

Reply:
yeah

Reply:
It might be a little confusing for users if `torch.ops.namespace.op_a` works but `torch.ops.namespace.op_a.overload` doesn't.  What do you think?

Reply:
We intend to use the `OpOverload` API in torch_dispatch, so it comes down to whether or not that's composable with TorchScript. In the current state, I think it would be ok to not add the TorchScript support.

Reply:
Alright, if it is not intended for end users we can skip TorchScript support. What do you think about improving the error message? Maybe we should add a special case there to let users know that per-overload op invocation isn't supported in TorchScript? The current message isn't clear or actionable enough for them. What do you think?